src/Core/Bridge/Symfony/Bundle/Command/UpgradeApiResourceCommand.php

```diff
@@ -103,6 +105,10 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         foreach ($this->resourceNameCollectionFactory->create() as $resourceClass) {
             try {
                 $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
+
+                if ($input->getArgument('fqcn') && $input->getArgument('fqcn') !== $resourceClass) {
```

Comment:
we can do that before calling `$this->resourceMetadataFactory->create($resourceClass);`